src/core/ext/transport/chttp2/server/chttp2_server.cc

```diff
@@ -206,10 +208,9 @@ class Chttp2ServerListener : public Server::ListenerInterface {
     // is created.
     grpc_chttp2_transport* transport_ ABSL_GUARDED_BY(&mu_) = nullptr;
     grpc_closure on_close_;
-    grpc_timer drain_grace_timer_;
-    grpc_closure on_drain_grace_time_expiry_;
-    bool drain_grace_timer_expiry_callback_pending_ ABSL_GUARDED_BY(&mu_) =
-        false;
+    absl::optional<EventEngine::TaskHandle> drain_grace_timer_handle_
+        ABSL_GUARDED_BY(&mu_);
+    EventEngine* event_engine_ ABSL_GUARDED_BY(&mu_);
```

Comment:
How do we guarantee the lifetime of this EventEngine object?

Reply:
Good callout! This event_engine_ is grabbed from a copy of the listener_'s ChannelArgs. And it also held a `RefCounterPtr<Chttp2ServerListener> listener_;` at line 202. Added a comment to clarify that.

Reply:
Slightly incorrect. We make a copy of the listener args, modify it and then pass it to `ActiveConnection`. I would suggest getting the pointer from the channel args in the listener and then explicitly pass the EventEngine pointer as a parameter to the constructor of ActiveConnection.

Reply:
Makes sense! Done.